src/Modules/Station/Components/OnlineUsers/OnlineUsers.tsx

```diff
@@ -85,50 +49,121 @@ export class OnlineUsersComponent extends BaseComponent<IProps, IState> {
+    if (id && currentUser) {
+      return currentUser.id === id;
     }
     return false;
   };
 
-  public renderPopoverContent() {
-    const filteredUsers = fixture.sort(
-      user => (user.username === currentUser.username ? 1 : 0),
+  public renderPopoverContent(onlineUsers: any[], numberOnline: 0) {
+    const { userInfo: currentUser } = this.props;
+
+    const filteredUsers = onlineUsers.sort(
+      user => (user.username === currentUser.username ? 0 : 1),
     );
+
+    // Only display keep top 10 users
+    const filteredList = filteredUsers.filter(
+      (user, index) => index < LIST_USERS_SHOW,
+    );
+    // Group the rest people to anonymous group
+    const anonymousUsersCount = numberOnline - filteredList.length;
+
     return (
       <ListGroup className="popover-container">
-        {filteredUsers.map(
+        {filteredList.map(
           ({ id, name, username, avatarUrl, points }, index) => (
             <Link to={`/profile/${id}`} key={index}>
               <ListGroupItem
-                active={this.isMe(username)}
+                active={this.isCurrentUser(id)}
                 className="online-users-list-item">
-                <div className="online-users-shape">
-                  <img
-                    className="online-users-image"
-                    alt="avatar"
-                    src={avatarUrl || DEFAULT_USER_AVATAR}
-                  />
-                </div>
+                <img
+                  className="online-users-image"
+                  alt="avatar"
+                  src={avatarUrl || DEFAULT_USER_AVATAR}
+                />
                 <span className="online-users-caption">
-                  {this.isUserInfoAvailable(currentUser) && this.isMe(username)
-                    ? `You (${points})`
-                    : `${name} (${points})`}
+                  {this.isUserInfoAvailable(currentUser) && this.isCurrentUser(id)
+                    ? `You (${points || 0})`
+                    : `${reduceByCharacters(name) || 'Unknown'} (${points ||
+                        0})`}
                 </span>
               </ListGroupItem>
             </Link>
           ),
         )}
+        {anonymousUsersCount > 0 ? (
+          <ListGroupItem className="online-users-list-item">
+            <div className="online-users-anonymous">
+              <span className="online-users-anonymous-text">
+                {anonymousUsersCount}
+              </span>
+            </div>
+            <span className="online-users-caption">Anonymous</span>
+          </ListGroupItem>
+        ) : null}
       </ListGroup>
     );
   }
 
+  public renderOnlineTooltip(
+    onlineUsers: any[],
+    numberOnline: 0,
+    target: string,
+  ) {
+    const { userInfo: currentUser } = this.props;
+
+    const filteredUsers = onlineUsers.sort(
+      user => (user.username === currentUser.username ? 0 : 1),
+    );
+    // Only display keep top 10 users
+    const filteredList = filteredUsers.filter(
```

Comment:
filteredList does not make sense. I suggest filteredListUserOnline.
Please check for filteredListToArray, filteredListWithoutAnonymous  as well